File: server/js-server/src/core/verbs.js
const fieldNames = [
    'id',
    'English',
    'Infinitive',
    'Participle_Past',
    'Gerund_Present',

    'Present_io',
    'Present_tu',
    'Present_lei',
    'Present_noi',
    'Present_voi',
    'Present_loro',

    'Imperfect_io',
    'Imperfect_tu',
    'Imperfect_lei',
    'Imperfect_noi',
    'Imperfect_voi',
    'Imperfect_loro',

    'Future_io',
    'Future_tu',
    'Future_lei',
    'Future_noi',
    'Future_voi',
    'Future_loro',

    'Past_io',
    'Past_tu',
    'Past_lei',
    'Past_noi',
    'Past_voi',
    'Past_loro',

    'Imperative_io',
    'Imperative_tu',
    'Imperative_lei',
    'Imperative_noi',
    'Imperative_voi',
    'Imperative_loro',

    'Conditional_io',
    'Conditional_tu',
    'Conditional_lei',
    'Conditional_noi',
    'Conditional_voi',
    'Conditional_loro',

    'Present_Subjunctive_io',
    'Present_Subjunctive_tu',
    'Present_Subjunctive_lei',
    'Present_Subjunctive_noi',
    'Present_Subjunctive_voi',
    'Present_Subjunctive_loro',

    'Imperfect_Subjunctive_io',
    'Imperfect_Subjunctive_tu',
    'Imperfect_Subjunctive_lei',
    'Imperfect_Subjunctive_noi',
    'Imperfect_Subjunctive_voi',
    'Imperfect_Subjunctive_loro',
]

const verbFields = new Set()

for (const v of fieldNames) {
    verbFields.add(v)
}

exports.verbFields = verbFields

File: server/go-server/src/graphql/verbs.go
package graphql

import (
	c "src/core"
	mysql "src/mysql"

	"github.com/graphql-go/graphql"
)

// func Hello() *graphql.Field {
// 	return &graphql.Field{
// 		Type: graphql.String,
// 		Resolve: func(p graphql.ResolveParams) (interface{}, error) {
// 			return "world", nil
// 		},
// 	}
// }

func FieldVerbs(ctx c.Ctx) *graphql.Field {
	var verbType = graphql.NewObject(graphql.ObjectConfig{
		Name: "verbs",
		Fields: graphql.Fields{
			"Id":                         &graphql.Field{Type: graphql.Int},
			"Infinitive":            &graphql.Field{Type: graphql.String},
			"Participle_Past":            &graphql.Field{Type: graphql.String},
			"Gerund_Present":             &graphql.Field{Type: graphql.String},
			"English":                    &graphql.Field{Type: graphql.String},
			"Present_io":                 &graphql.Field{Type: graphql.String},
			"Present_tu":                 &graphql.Field{Type: graphql.String},
			"Present_lei":                &graphql.Field{Type: graphql.String},
			"Present_noi":                &graphql.Field{Type: graphql.String},
			"Present_voi":                &graphql.Field{Type: graphql.String},
			"Present_loro":               &graphql.Field{Type: graphql.String},
			"Imperfect_io":               &graphql.Field{Type: graphql.String},
			"Imperfect_tu":               &graphql.Field{Type: graphql.String},
			"Imperfect_lei":              &graphql.Field{Type: graphql.String},
			"Imperfect_noi":              &graphql.Field{Type: graphql.String},
			"Imperfect_voi":              &graphql.Field{Type: graphql.String},
			"Imperfect_loro":             &graphql.Field{Type: graphql.String},
			"Future_io":                  &graphql.Field{Type: graphql.String},
			"Future_tu":                  &graphql.Field{Type: graphql.String},
			"Future_lei":                 &graphql.Field{Type: graphql.String},
			"Future_noi":                 &graphql.Field{Type: graphql.String},
			"Future_voi":                 &graphql.Field{Type: graphql.String},
			"Future_loro":                &graphql.Field{Type: graphql.String},
			"Past_io":                    &graphql.Field{Type: graphql.String},
			"Past_tu":                    &graphql.Field{Type: graphql.String},
			"Past_lei":                   &graphql.Field{Type: graphql.String},
			"Past_noi":                   &graphql.Field{Type: graphql.String},
			"Past_voi":                   &graphql.Field{Type: graphql.String},
			"Past_loro":                  &graphql.Field{Type: graphql.String},
			"Imperative_io":              &graphql.Field{Type: graphql.String},
			"Imperative_tu":              &graphql.Field{Type: graphql.String},
			"Imperative_lei":             &graphql.Field{Type: graphql.String},
			"Imperative_noi":             &graphql.Field{Type: graphql.String},
			"Imperative_voi":             &graphql.Field{Type: graphql.String},
			"Imperative_loro":            &graphql.Field{Type: graphql.String},
			"Conditional_io":             &graphql.Field{Type: graphql.String},
			"Conditional_tu":             &graphql.Field{Type: graphql.String},
			"Conditional_lei":            &graphql.Field{Type: graphql.String},
			"Conditional_noi":            &graphql.Field{Type: graphql.String},
			"Conditional_voi":            &graphql.Field{Type: graphql.String},
			"Conditional_loro":           &graphql.Field{Type: graphql.String},
			"Present_Subjunctive_io":     &graphql.Field{Type: graphql.String},
			"Present_Subjunctive_tu":     &graphql.Field{Type: graphql.String},
			"Present_Subjunctive_lei":    &graphql.Field{Type: graphql.String},
			"Present_Subjunctive_noi":    &graphql.Field{Type: graphql.String},
			"Present_Subjunctive_voi":    &graphql.Field{Type: graphql.String},
			"Present_Subjunctive_loro":   &graphql.Field{Type: graphql.String},
			"Imperfect_Subjunctive_io":   &graphql.Field{Type: graphql.String},
			"Imperfect_Subjunctive_tu":   &graphql.Field{Type: graphql.String},
			"Imperfect_Subjunctive_lei":  &graphql.Field{Type: graphql.String},
			"Imperfect_Subjunctive_noi":  &graphql.Field{Type: graphql.String},
			"Imperfect_Subjunctive_voi":  &graphql.Field{Type: graphql.String},
			"Imperfect_Subjunctive_loro": &graphql.Field{Type: graphql.String},
		},
	})

	return &graphql.Field{
		Type: graphql.NewList(verbType),
		Args: graphql.FieldConfigArgument{
			"id": &graphql.ArgumentConfig{
				Type:         graphql.Int,
				DefaultValue: nil,
			},
			"infinitive": &graphql.ArgumentConfig{
				Type:         graphql.String,
				DefaultValue: nil,
			},
			"searchTerm": &graphql.ArgumentConfig{
				Type:         graphql.String,
				DefaultValue: nil,
			},
		},
		Resolve: func(p graphql.ResolveParams) (interface{}, error) {
			return mysql.SearchVerbs(ctx, p)
		},
	}
}
